### ISSUE TITLE
Server process terminates with no debug messages on Complete Order

### ISSUE THREAD

**Issue by Git909Monk** [OpenBazaar/OpenBazaar-Server#365]
Brief Description:  When trying to compete an order the server process goes belly up without so much as a moan. This is like issue #347 but there is no related message with loglevel debug.

Operating System (OS and version):  RHEL 7.2
OpenBazaar version:  pulled 4-20-2015
Hardware: ESXi guest

Reproducible (Always / Almost Always / Sometimes / Rarely / Couldn't Reproduce):  Still can't un-reproduce. Will instruct moderator to release funds (right?) by end of day as this is not a trivial amount. 

Steps to reproduce:
1. Type in review
2. Hit Save
3. pull hair

Observed Behavior:
Leaving the Review field blank will error with missing data message.

Expected Behavior:
Completing the contract.

Additional info (links, images, etc go here):
I randomly get (mostly) pubkey error messages if I try to create a contract. This is one of the few contracts I was able to initiate.

**Comment by Git909Monk** [comment]
Command line: /usr/bin/python openbazaard.py start -d -a 12.34.56.789 -l debug

**Comment by hoffmabc** [comment]
Please take a look at this one @cpacia

**Comment by Git909Monk** [comment]
Looks like the last packets received are for POST /api/v1/complete_order HTTP/1.1
I have a pcap I can supply if it would be helpful.

**Comment by cpacia** [comment]
Yeah if you can give me a copy of contract json I can look to see what might have caused it. 

It would be helpful too if you can get the seller to send his as well. If I have both I should be able to figure out the problem. You can either post it here or email to me ctpacia@gmail.com. I also have a pgp key you need it.

**Comment by Git909Monk** [comment]
You should have received the files yesterday. The contract dispute completed several hours ago. My contract then updated to show there was a dispute and that is was completed. The clients button changed to Accept Dispute Payout. Clicking this also crashes the server quickly and quietly. I might have an updated client, but this time the client died with a white screen when trying to reconnect. Had to restart the client.

Guess this contract will be forever in limbo for me. Don't know if there was a 'payout'. Hope my next contract works better...

**Comment by cpacia** [comment]
I don't see this email in my inbox or junk folder. Do you know what the subject was so I can search for it?

**Comment by Git909Monk** [comment]
the subject was "Files for issue #365" to ctpacia@gmail.com Date: Fri, 22 Apr 2016 05:21:19 -0700

**Comment by cpacia** [comment]
got it

**Comment by cpacia** [comment]
OK it's showing that the funds were successfully released from the multisig to the vendor. 

https://live.blockcypher.com/btc/address/32je67qTUZoUXZLG7ZBuhW5FZBWZ7WEKPx/

I can't tell from the data I have if that was the result of you're node broadcasting the transaction or if it was a moderator payout. But it doesn't look like the funds are locked up any more.

**Comment by Git909Monk** [comment]
That is good. Anyway for me to debug the server crash?

**Comment by cpacia** [comment]
If you can look in the OB-Mainnet.db database under the `purchases` table... take a look at what the `outpoints` column says for your order ID. 

The only thing I can think of is that your client might not have detected the bitcoin payment properly. 

Also make sure you are using the latest version. There was a similar bug that I fixed recently.

**Comment by Git909Monk** [comment]
The initial payment went fine. The problem occurred when I tried to "Complete Order", and latter at "Accept Dispute Payout" in the client.

**Comment by cpacia** [comment]
Yeah the "Complete Order" button actually builds a bitcoin transaction and releases the funds. If there was some kind of issue saving the payment details it could possibly fail to build the payout tx which would cause the complete order to fail. Not sure if that's what happened here.

**Comment by Git909Monk** [comment]
Updated file and purchases.outpoint data send.

**Comment by cpacia** [comment]
Hmm.. that looks normal. When you clicked complete order did the UI display an error message  (if so what did it say?) or was it just crashing? 

If it was just a crash it makes me think it might have been something to do with the environment.

**Comment by Git909Monk** [comment]
The UI goes to server setup window as soon as the button is pressed.Same for the Accept Dispute Payout button.

**Comment by Git909Monk** [comment]
This issue still exists. The guid was moved to  another server and when the client connects it operates normally and does not crash the server process. I have reached a dead end trying to trouble shoot this issue. Sodium was not installed initially and installing it had no effect so I assume that is no longer a requirement?
